src/types/types.ts

```diff
@@ -1840,6 +1840,12 @@ export interface EdgeContext {
   readonly forgetAccount: (rootLoginId: string) => Promise<void>
 
   // Account creation:
+  /** Preemptively requests a CAPTCHA for account creation. */
+  readonly fetchChallenge: () => Promise<{
+    challengeId: string
+    /** If this is missing, the challenge is already solved. */
```

Comment:
Maybe an explicit status would make this more clear without having to read prop documentation

Reply:
I had it this way originally, but changed it this way for type-safety. You literally can't show the challenge to a user if there is no URL - what would go in the modal?